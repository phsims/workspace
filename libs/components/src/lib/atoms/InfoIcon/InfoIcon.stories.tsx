import type { Meta, StoryObj } from '@storybook/react';
import RestaurantRoundedIcon from '@mui/icons-material/RestaurantRounded';

import { InfoIcon } from './InfoIcon';

import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

const meta: Meta<typeof InfoIcon> = {
  component: InfoIcon,
  title: '@atoms/InfoIcon',
};
export default meta;
type Story = StoryObj<typeof InfoIcon>;

export const Primary = {
  args: {
    icon: <RestaurantRoundedIcon fontSize="medium" color="primary" />,
    infoTitle: 'Serves:',
    text: '4',
  },
};

export const Heading: Story = {
  args: { ...Primary.args },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Welcome to InfoIcon!/gi)).toBeInTheDocument();
  },
};
