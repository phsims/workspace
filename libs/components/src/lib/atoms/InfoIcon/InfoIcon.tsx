import { Box, Typography } from '@mui/material';
import { ReactNode } from 'react';

export interface InfoIconProps {
  infoTitle: string;
  icon: ReactNode;
  text: string;
}

export function InfoIcon({ infoTitle, icon, text }: InfoIconProps) {
  return (
    <Box sx={{ display: 'flex', flexDirection: 'row', gap: '0.5rem' }}>
      {icon}
      <Typography variant="body1_Bold" color={'grey.dark'}>
        {infoTitle}
      </Typography>
      <Typography variant="body1" color={'grey.dark'}>
        {text}
      </Typography>
    </Box>
  );
}

export default InfoIcon;
