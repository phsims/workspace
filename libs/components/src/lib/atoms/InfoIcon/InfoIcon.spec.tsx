import { render } from '@testing-library/react';
import RestaurantRoundedIcon from '@mui/icons-material/RestaurantRounded';

import InfoIcon from './InfoIcon';

const mockProps = {
  icon: <RestaurantRoundedIcon fontSize="medium" color="primary" />,
  infoTitle: 'Serves:',
  text: '4',
};

describe('InfoIcon', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<InfoIcon {...mockProps} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = render(<InfoIcon {...mockProps} />);
    expect(baseElement).toMatchSnapshot();
  } );

  it('should render the icon', () => {
    const { getByTestId } = render(<InfoIcon {...mockProps} />);
    expect(getByTestId('RestaurantRoundedIcon')).toBeInTheDocument();
  });
});
