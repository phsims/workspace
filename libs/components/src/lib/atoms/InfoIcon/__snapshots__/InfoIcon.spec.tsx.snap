// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoIcon should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1f0wxn3"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        data-testid="RestaurantRoundedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16 6v6c0 1.1.9 2 2 2h1v7c0 .55.45 1 1 1s1-.45 1-1V3.13c0-.65-.61-1.13-1.24-.98C17.6 2.68 16 4.51 16 6m-5 3H9V3c0-.55-.45-1-1-1s-1 .45-1 1v6H5V3c0-.55-.45-1-1-1s-1 .45-1 1v6c0 2.21 1.79 4 4 4v8c0 .55.45 1 1 1s1-.45 1-1v-8c2.21 0 4-1.79 4-4V3c0-.55-.45-1-1-1s-1 .45-1 1z"
        />
      </svg>
      <span
        class="MuiTypography-root MuiTypography-body1_Bold css-yylahi-MuiTypography-root"
      >
        Serves:
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-4z0fiv-MuiTypography-root"
      >
        4
      </p>
    </div>
  </div>
</body>
`;
